### PR TITLE
added functionality, which removes untracked local branches

### PR DESCRIPTION
**Pull request by tdanylchuk**
On behalf of:
https://github.com/spring-cloud/spring-cloud-config/issues/877

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@tdanylchuk Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=940)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=940) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by ryanjbaxter** [comment]
@tdanylchuk this looks pretty good, could you add some documentation for the new property?

**Comment by tdanylchuk** [comment]
Yes... Forgot it, will surely add it, anything else to update?

**Comment by tdanylchuk** [comment]
@ryanjbaxter I've update doc, please take a look.

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java [hunk 1]
```diff
@@ -16,6 +16,8 @@
 package org.springframework.cloud.config.server.environment;
 
 
+import org.apache.commons.lang3.ArrayUtils;
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: Please don't use commons-lang

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java [hunk 2]
```diff
@@ -90,7 +101,7 @@ public static JGitConfigServerTestData prepareClonedGitRepository(Class... sourc
         //setup local repository
         File clonedRepoDir = new File("target/repos/cloned");
         if(clonedRepoDir.exists()) {
-            FileSystemUtils.deleteRecursively(clonedRepoDir);
+            FileUtils.deleteDirectory(clonedRepoDir);
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: Please revert to `FileSystemUtils.deleteRecursively`

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java [hunk 3]
```diff
@@ -102,8 +113,11 @@ public static JGitConfigServerTestData prepareClonedGitRepository(Class... sourc
                 .call();
 
         //setup our test spring application pointing to the local repo
+        String[] properties = additionalProperties.toArray(new String[0]);
+        properties = ArrayUtils.add(properties, "spring.cloud.config.server.git.uri:" + "file://" + clonedRepoDir.getAbsolutePath());
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: Don't use `ArrayUtils`. Clone `additionalProperties` to a new list and add a new property, then convert to array.

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java [hunk 4]
```diff
@@ -479,16 +480,6 @@ private String getCommitID(Git git, String label) throws GitAPIException {
 		return localRef.getObjectId().getName();
 	}
 
-	public void passphrase() throws IOException {
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java]: Why was this test removed?

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java (hunk 2 above)
Comment (tdanylchuk) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: ok, didn't work for my windows ...

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java (hunk 4 above)
Comment (tdanylchuk) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java]: method was unused

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java (hunk 1 above)
Comment (tdanylchuk) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: done

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java (hunk 3 above)
Comment (tdanylchuk) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitConfigServerTestData.java]: done

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java (hunk 4 above)
Comment (tdanylchuk) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java]: reverted

**Comment by spencergibb** [comment]
Closed with some polish via fffbb644cbd5b56225ecc59624c7b74254430d37

**Pull request closed without merging**